File: project/server/src/main/java/ru/zagrebin/dto/PostFullDto.java
package ru.zagrebin.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.List;

@Data
public class PostFullDto {
    private Long id;
    private String postType;
    private String status;

    private String title;
    private String excerpt;
    private String content;

    private String coverUrl;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;

    private AuthorShortDto author;

    private List<TagDto> tags;
    private List<PostIngredientDto> ingredients;
    private List<RecipeStepDto> steps;

    private int likesCount;
    @JsonProperty("liked")
    private boolean liked;
    private long viewsCount;

    private Integer calories;
    private Integer cookingTimeMinutes;
}

File: project/mobile/app/src/main/java/ru/zagrebin/culinaryblog/data/remote/dto/PostFullDto.kt
package ru.zagrebin.culinaryblog.data.remote.dto

data class PostFullDto(
    val id: Long,
    val postType: String?,
    val status: String?,
    val title: String?,
    val excerpt: String?,
    val content: String?,
    val coverUrl: String?,
    val createdAt: String?,
    val updatedAt: String?,
    val author: PostAuthorDto?,
    val tags: List<PostTagDto>?,
    val ingredients: List<PostIngredientLineDto>?,
    val steps: List<PostStepDto>?,
    val likesCount: Int?,
    val liked: Boolean?,
    val viewsCount: Long?,
    val calories: Int?,
    val cookingTimeMinutes: Int?
)
